### PR TITLE
IGNITE-17357 JMX metrics exporter

### PR DESCRIPTION
**Pull request by kgusakov**
https://issues.apache.org/jira/browse/IGNITE-17357

### CODE REVIEW

File: modules/metrics/src/main/java/org/apache/ignite/internal/metrics/exporters/jmx/JmxExporter.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    /**
+     * Logger.
+     */
+    private static IgniteLogger LOG = Loggers.forClass(JmxExporter.class);
+
+    /**
+     * Current registered MBeans.
+     */
+    private final List<ObjectName> mbeans = new ArrayList<>();
```
Comment (denis-chudov) [inline modules/metrics/src/main/java/org/apache/ignite/internal/metrics/exporters/jmx/JmxExporter.java:59]: looks like nothing protects us from attempts of concurrent modification of this collection, are you sure that ArrayList is suitable here?

File: modules/metrics/src/main/java/org/apache/ignite/internal/metrics/exporters/jmx/JmxExporter.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            } else {
+                LOG.warn("Tried to unregister the MBean for non-registered metric set " + metricSetName);
+            }
+        } catch (MalformedObjectNameException e) {
+            LOG.error("MBean for metric set " + metricSetName + " can't be unregistered.", e);
+        }
+    }
+
+    /**
+     * Unregister MBean by its' name.
```
Comment (denis-chudov) [inline modules/metrics/src/main/java/org/apache/ignite/internal/metrics/exporters/jmx/JmxExporter.java:154]: typo?

File: modules/metrics/src/main/java/org/apache/ignite/internal/metrics/MetricManager.java [hunk 3]
```diff
@@ -178,15 +192,19 @@ public static Map<String, MetricExporter> loadExporters() {
...
     /**
-     * Disable metric source by name. See {@link MetricRegistry#disable(String)}.
+     * Disable metric set by name. See {@link MetricRegistry#disable(String)}.
      *
-     * @param srcName Source name.
+     * @param metricSetName Metric set name.
      */
-    public void disable(final String srcName) {
-        registry.disable(srcName);
+    public void disable(final String metricSetName) {
```
Comment (denis-chudov) [inline modules/metrics/src/main/java/org/apache/ignite/internal/metrics/MetricManager.java:204]: I don't completely understand this change. It refers to `MetricRegistry#disable(String)` but it disables metric source by its name. Also, it brings inconsistency to API, as `MetricManager#enable(String)` enables metric source, not metric set. Finally, only metric source can be enabled/disabled. Could you explain?

File: modules/metrics/src/main/java/org/apache/ignite/internal/metrics/exporters/jmx/JmxExporter.java (hunk 1 above)
Comment (kgusakov) [reply modules/metrics/src/main/java/org/apache/ignite/internal/metrics/exporters/jmx/JmxExporter.java:59]: This private collection is supposed to use only from synchronized methods

File: modules/metrics/src/main/java/org/apache/ignite/internal/metrics/exporters/jmx/JmxExporter.java (hunk 2 above)
Comment (kgusakov) [reply modules/metrics/src/main/java/org/apache/ignite/internal/metrics/exporters/jmx/JmxExporter.java]: fixed

File: modules/metrics/src/main/java/org/apache/ignite/internal/metrics/MetricManager.java (hunk 3 above)
Comment (kgusakov) [reply modules/metrics/src/main/java/org/apache/ignite/internal/metrics/MetricManager.java]: Renamed by mistake. Fixed, thanks.

**Pull request merged**